tests/gauge_force_test.cpp

```diff
@@ -214,6 +236,127 @@ void gauge_force_test(bool compute_force = true)
   delete Mom_ref_milc;
 }
 
+void gauge_loop_test()
+{
+  int max_length = 6;
+
+  setVerbosityQuda(QUDA_VERBOSE, "", stdout);
+
+  QudaGaugeParam gauge_param = newQudaGaugeParam();
+  setGaugeParam(gauge_param);
+
+  gauge_param.gauge_order = gauge_order;
+  gauge_param.t_boundary = QUDA_PERIODIC_T;
+
+  setDims(gauge_param.X);
+
+  double *trace_loop_coeff_p = &trace_loop_coeff_d[0];
+  int num_paths = sizeof(trace_path) / sizeof(trace_path[0]);
+  int *trace_loop_length_p = &trace_loop_length[0];
+
+  // b/c we can't cast int*[6] -> int**
+  int** trace_path_p = new int*[num_paths];
```

Comment:
Could make this a vector of vectors to remove the `delete` junk at the end

Reply:
Same issue as before, but I guess there's no harm in making everything use a `std::vector<...>`. Will do in a subsequent commit.

Reply:
Discussed offline that, because the QUDA interface needs to be C-friendly, there's no point in wrapping this in a `std::vector` just to unwrap it before passing it to a function.